### PR TITLE
Remove table flush on destruction of PagedArray

### PR DESCRIPTION
**Pull request by juliantaylor**
The PagedArray destructor flushes its underlying Table which essentially
makes the write Tile cache useless as many short lived objects do
contain PagedArray objects. When iterating over a Lattice in an
order that does not match the tileshape it will flush the cache on
destructions of for example LatticeIterators, LatticeExpressions or
SubImages and it while thus constantly rewrite partially modified tiles.
In some CASA tasks this leads to unnecessary very large write
amplifications.

To remove the all code that explicitly looks at table files on disk must
be adapted to respect the TableCache as it the flush frequency is
massively reduced and the files might not exist on disk at the time the
functions try to read them.
This commit updates Table::isReadable and TableInfo which is enough to
make most of the CASA regression suite succeed.
This also requires removing an unnecessary lock from
TableCache::lookCache which is during the reopenRW call may otherwise
recursively lock the TableCache.

IsWritable is not changed as it does not appear necessary currently and
if would need additional logic to handle read-only tables in cache that
are supposed to be opened in write mode.

closes gh-453

### CODE REVIEW

**Comment by juliantaylor** [comment]
@tammojan can you please test this branch, it is a large behaviour change and is quite likely to break stuff.

File: tables/Tables/TableCache.cc [hunk 1]
```diff
@@ -170,9 +170,8 @@ void TableCache::flushTable (const String& name,
 PlainTable* TableCache::lookCache (const String& name, int tableOption,
                                    const TableLock& lockOptions)
 {
-    ScopedMutexLock sc(itsMutex);
```
Comment (tammojan) [inline tables/Tables/TableCache.cc]: Why did you delete the mutexes?

Comment (juliantaylor) [inline tables/Tables/TableCache.cc]: the mutex is not necessary for the full function, we only need it to access the shared state, the `tableMap_p` which is done in the operator call following it.
if we keep this mutex we would need recursive locking as the reopenRW will call isReadable which calls back into the cache.

Comment (juliantaylor) [inline tables/Tables/TableCache.cc]: or at least it appears to be the only state, you would assume that the synchronization of the calls into the shared table object would be handled by the lock of he table.
But it doesn't seem to have a lock so maybe we need to make it recursive ..

**Comment by juliantaylor** [comment]
Only part of the casa test suite is capable to run from a git branch, so to test the rest a merge is simplest way.
Should problems show up please let me know, I will follow up with fixes or a revert if necessary. I am also open to changing the locking scheme to recursive as it might be a practicality over purity case.
